### PR TITLE
New ROI commands implementation

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
Replacement for #8683 based on latest master.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
From @DonLakeFlyer:

@LorenzMeier @ndepal 

I just testing this pull rebased to current master using SITL Gazebo. The plan just does and ROI_LOCATION for a couple waypoints and then an ROI_NONE.
[ROITesting.plan.zip](https://github.com/PX4/Firmware/files/1653234/ROITesting.plan.zip)

MIS_MNT_YAW_CTL = 0
* Correct: Vehicle yawed correctly to point to ROI_LOCATION at correct mission item
* Incorrect: At ROI_LOCATION item, gimbal pointed straight down (or close to that)
* Correct: Vehicle went back to pointing along waypoint path at ROI_NONE

MIS_MNT_YAW_CTL = 1
* Correct: Vehicle did not yaw at ROI_LOCATION item
* Incorrect: Gimbal pitched down at what looked like 45 degrees, did not yaw at ROI_LOCATION
* Correct: Vehicle went back to pointing along waypoint path at ROI_NONE

**Comment by LorenzMeier** [comment]
@DonLakeFlyer I had a look but I won't be able to finish this tonight. When you say it pointed straight down, why was that incorrect? When it was at the ROI location shouldn't it pitch straight down to have the ROI in view?

@bkueng If you're on the job tomorrow: Could you look into adding VEHICLE_CMD_DO_SET_ROI_WPNEXT_OFFSET ? The intent is to add the angles it carries to the current heading. So if it says yaw 90 then it should yaw 90 degrees in addition to whatever the current vehicle mode configured it to (e.g. towards next waypoint).

@ndepal Could you look into the pitch transformation in the PR?

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -1176,6 +1176,9 @@ Commander::handle_command(vehicle_status_s *status_local, const safety_s *safety
 	case vehicle_command_s::VEHICLE_CMD_NAV_DELAY:
 	case vehicle_command_s::VEHICLE_CMD_DO_SET_ROI:
 	case vehicle_command_s::VEHICLE_CMD_NAV_ROI:
+    case vehicle_command_s::VEHICLE_CMD_DO_SET_ROI_LOCATION:
```
Comment (ndepal) [inline src/modules/commander/commander.cpp]: indentation

File: src/modules/navigator/navigation.h [hunk 2]
```diff
@@ -74,6 +74,9 @@ enum NAV_CMD {
 	NAV_CMD_DO_SET_HOME = 179,
 	NAV_CMD_DO_SET_SERVO = 183,
 	NAV_CMD_DO_LAND_START = 189,
+	NAV_CMD_DO_SET_ROI_LOCATION = 195,
+	NAV_CMD_DO_SET_ROI_WPNEXT_OFFSET = 196,
+	NAV_CMD_DO_SET_ROI_NONE = 197,
 	NAV_CMD_DO_SET_ROI = 201,
```
Comment (ndepal) [inline src/modules/navigator/navigation.h]: Is `NAV_CMD_DO_SET_ROI` still needed?

**Comment by ndepal** [comment]
>Incorrect: At ROI_LOCATION item, gimbal pointed straight down (or close to that)

This sounds like the relative alt problem. It seems that my fix may have been broken with the new command names. https://github.com/DonLakeFlyer/Firmware/blob/954e4a049468d811499c54b6e65bfefe131e0da7/src/modules/navigator/mission_block.cpp#L455

**Comment by LorenzMeier** [comment]
@ndepal Could you please do the change you suggested or let us know if we should close it until you've wrapped up?

**Comment by LorenzMeier** [comment]
@bkueng Your help is appreciated for the new commands

**Comment by ndepal** [comment]
Fixed the nav cmd. Tested in SITL.

**Pull request merged**